### PR TITLE
PYTHON-4409 Streamline Wheel Building

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: .github/workflows/release-python.yml [hunk 1]
```diff
@@ -29,43 +29,25 @@ jobs:
...
-            - buildplat: [macos-11, macosx_*]
-              python: "cp38"
-            - buildplat: [macos-11, macosx_*]
-              python: "cp39"
-            - buildplat: [macos-14, macosx_*]
-              python: "cp310"
-            - buildplat: [macos-14, macosx_*]
-              python: "cp311"
-            - buildplat: [macos-14, macosx_*]
-              python: "cp312"
```
Comment (Jibola) [inline .github/workflows/release-python.yml:52]: Why can we get away with not specifying the python version anymore Is It that the way we build wheels now will automatically create wheels packages for 3.7 - 3.12?

File: .github/workflows/release-python.yml [hunk 2]
```diff
@@ -79,75 +61,37 @@ jobs:
...
-          python -m pip install "cibuildwheel>=2.4,<3"
+          python -m pip install "cibuildwheel>=2.17,<3"
 
-      - name: Build wheel
-        if: ${{ matrix.buildplat[0] != 'macos-14' }}
+      - name: Build wheels
         env:
-          CIBW_BUILD: ${{ matrix.python }}-${{ matrix.buildplat[1] }}
-          MACOS_TEST_SKIP: "*arm64"
+          CIBW_BUILD: ${{ matrix.buildplat[2] }}
```
Comment (Jibola) [inline .github/workflows/release-python.yml:68]: Should this step exclude manylinux1?

**Review by Jibola — APPROVED** [review]
LGTM. 
Left questions just to improve my understanding.

File: .github/workflows/release-python.yml (hunk 1 above)
Comment (blink1073) [reply .github/workflows/release-python.yml:52]: Yes, and we verify it later. :)

File: .github/workflows/release-python.yml (hunk 2 above)
Comment (blink1073) [reply .github/workflows/release-python.yml:68]: No need to exclude it.  We have a separate manylinux1 to ensure that we have wheels that include that arch, but it is fine if we have duplicate wheels.

**Pull request merged**